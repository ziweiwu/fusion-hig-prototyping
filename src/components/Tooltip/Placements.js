import positions from 'positions';

//set autoAdjustOverflow options
const autoAdjustOverflowEnabled = {
  adjustX: 1,
  adjustY: 1,
};

const autoAdjustOverflowDisabled = {
  adjustX: 0,
  adjustY: 0,
};

export function getOverflowOptions(autoAdjustOverflow) {
  if (typeof autoAdjustOverflow === 'boolean') {
    return autoAdjustOverflow ? autoAdjustOverflowEnabled : autoAdjustOverflowDisabled;
  }
  return {
    ...autoAdjustOverflowDisabled,
    ...autoAdjustOverflow,
  };
}

// fix the issue when set overflow, the arrow position is inaccurate
// https://github.com/react-component/tooltip/issues/54

// const placementsMap = {
//   tc: 'top center',
//   bc: 'bottom center',
//   cl: 'center right',
//   cr: 'center left',
// };
//
// export function placeArrow(tooltipEl, align) {
//   const arrowEl = tooltipEl.querySelector('.hig-tooltip-arrow');
//   const targetEl = this.getRootDomNode();
//   const position = positions(arrowEl, placementsMap[align.points[0]], targetEl, placementsMap[align.points[1]]);
//
//   if (align.points[0] === 'tc' || align.points[0] === 'bc') {
//     arrowEl.style.top = '';
//     arrowEl.style.left = `${position.left}px`;
//   } else {
//     arrowEl.style.top = `${position.top}px`;
//     arrowEl.style.left = '';
//   }
// }

