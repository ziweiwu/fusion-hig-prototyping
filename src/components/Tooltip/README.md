## Tooltip

Tooltips generally describe what a tool or feature does, or when or how to use it.
Read more about it on [Internal Wiki Page](https://hig.autodesk.com/web/components/tooltips).

### Props
- anchorPoint: specify placement of tooltip 
- trigger: specify the mean to trigger tooltip. Including "click", "trigger", and "focus"
- mouseEnterDelay: specify the duration before tooltip appear when mouse enters hover
- mouseLeaveDelay: specify the duration before tooltip disappear when mouse leaves hover
- title: a simple headline to describe tooltip
- description: a more detail instruction to expand on title
- content: additional HTML node can be added as content, for example an embedded picture or video  
- linkTitle: a hyperlink text points to the link URL. If not specified, the URL would be displayed instead
- linkURL: a hyperlink to a web address 
- isVisible: use prop to control tooltip popover visibility instead of component state 


