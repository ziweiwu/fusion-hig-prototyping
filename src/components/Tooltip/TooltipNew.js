import React from 'react';
import {Manager, Reference, Popper} from 'react-popper';
import Button from '@hig/button';
import '@hig/button/build/index.css';


export default class Example extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hide: true,
    };
    this.toggleHide = this.toggleHide.bind(this);
    this.onMouseEnter = this.onMouseEnter.bind(this);
    this.onMouseLeave = this.onMouseLeave.bind(this);
  }

  toggleHide() {
    this.setState((prevState) => ({ hide: !prevState.hide,
    }));
  }

  onMouseEnter(){
    this.setState(()=>({
      hide: true,
    }));
  }
  onMouseLeave(){
    this.setState(()=>({
      hide: false,
    }));
  }

  render() {
    const props = this.props;
    return (
      <Manager>
        <div>
          <Reference>
            {({ref}) => (
              <div ref={ref} onClick={this.toggleHide} /*onMouseEnter={this.onMouseEnter} onMouseLeave={this.onMouseLeave}*/>
              <Button type="primary" title="My button"/>
              </div>
            )}
          </Reference>
          {!this.state.hide &&
            <Popper placement="left">
              {({ref, style, placement, arrowProps}) => (
                <div ref={ref} style={style} data-placement={placement}>
                  {props.content}
                  <div ref={arrowProps.ref} style={arrowProps.style}/>
                </div>
              )}
            </Popper>
          }
        </div>
      </Manager>
    );
  }
}
