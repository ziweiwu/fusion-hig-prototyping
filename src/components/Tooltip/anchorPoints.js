export const anchorPoints = Object.freeze({
  TOP_CENTER: "top-center",
  RIGHT_CENTER: "right-center",
  BOTTOM_CENTER: "bottom-center",
  LEFT_CENTER: "left-center"
});

export const availableAnchorPoints = Object.freeze(Object.values(anchorPoints));
