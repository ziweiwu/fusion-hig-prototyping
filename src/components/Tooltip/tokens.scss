@import "~@hig/styles/mixins/colors.scss";
@import "~@hig/styles/mixins/spacing.scss";

$hig-font-family: ArtifaktElement, sans-serif !default;
$hig-tooltip-background-color: color("hig-gray-60");
$hig-tooltip-font-color: #ffffff;
$tooltip-border-radius: 2px;
$tooltip-padding: spacing("xxxs");
