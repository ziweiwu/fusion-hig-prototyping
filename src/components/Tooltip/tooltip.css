@import url(~@hig/styles/build/index.css);
.fade-enter {
  opacity: 0.01; }

.fade-enter.fade-enter-active {
  opacity: 1;
  transition: opacity 200ms ease-in; }

.fade-leave {
  opacity: 1; }

.fade-leave.fade-leave-active {
  opacity: 0.01;
  transition: opacity 200ms ease-in; }

/* theme variables */
.hig-tooltip {
  font-family: ArtifaktElement, sans-serif;
  position: relative;
  z-index: 1060;
  display: inline-block;
  list-style: none;
  visibility: visible;
  font-size: 12px;
  text-align: left;
  margin: 0;
  padding: 0; }
  .hig-tooltip button, .hig-tooltip select {
    font-family: ArtifaktElement, sans-serif; }

.hig-tooltip-placement-top, .hig-tooltip-placement-topLeft, .hig-tooltip-placement-topRight {
  padding: 8px 0 8px 0; }

.hig-tooltip-placement-right, .hig-tooltip-placement-rightTop, .hig-tooltip-placement-rightBottom {
  padding: 0 8px 0 8px; }

.hig-tooltip-placement-bottom, .hig-tooltip-placement-bottomLeft, .hig-tooltip-placement-bottomRight {
  padding: 8px 0 8px 0; }

.hig-tooltip-placement-left, .hig-tooltip-placement-leftTop, .hig-tooltip-placement-leftBottom {
  padding: 0 8px 0 8px; }

.hig-tooltip-children-wrapper {
  display: inline-block;
  position: relative; }

.hig-tooltip-arrow {
  position: absolute;
  border: solid transparent; }

.hig-tooltip-light .hig-tooltip-inner {
  color: black;
  background-color: white;
  border: 1px solid #D4DBE1; }
  .hig-tooltip-light .hig-tooltip-inner hr {
    background-color: lightgray;
    color: lightgray;
    border-color: lightgray; }
  .hig-tooltip-light .hig-tooltip-inner a {
    color: #393939; }
  .hig-tooltip-light .hig-tooltip-inner hr {
    background-color: #ffffff;
    color: #ffffff;
    border-color: #ffffff; }

.hig-tooltip-dark .hig-tooltip-inner {
  color: #FFFFFF;
  background-color: #393939;
  box-shadow: 0 1px 4px; }
  .hig-tooltip-dark .hig-tooltip-inner a {
    color: black; }
  .hig-tooltip-dark .hig-tooltip-inner hr {
    background-color: lightgray;
    color: lightgray;
    border-color: lightgray; }
  .hig-tooltip-dark .hig-tooltip-inner a {
    color: #fff; }
  .hig-tooltip-dark .hig-tooltip-inner hr {
    background-color: black;
    color: black;
    border-color: black; }

.hig-tooltip-inner {
  padding: 6px 10px;
  text-align: left;
  text-decoration: none;
  line-height: 12px;
  box-shadow: none;
  border-radius: 5px; }
  .hig-tooltip-inner img {
    max-width: 300px;
    height: auto; }
  .hig-tooltip-inner .hig-tooltip-link {
    text-align: right; }

.hig-tooltip-title {
  font-weight: bold;
  padding-top: 8px;
  padding-bottom: 8px; }

.hig-tooltip-description {
  font-weight: normal;
  padding-top: 8px;
  padding-bottom: 8px; }

/* inner arrow and animations */
.hig-tooltip-placement-top .hig-tooltip-arrow, .hig-tooltip-placement-topLeft .hig-tooltip-arrow, .hig-tooltip-placement-topRight .hig-tooltip-arrow {
  bottom: 4px;
  margin-left: -5px;
  border-width: 5px 5px 0;
  border-top-color: #393939; }

.hig-tooltip-placement-bottom .hig-tooltip-arrow, .hig-tooltip-placement-bottomLeft .hig-tooltip-arrow, .hig-tooltip-placement-bottomRight .hig-tooltip-arrow {
  top: 4px;
  margin-left: -5px;
  border-width: 0px 5px 5px;
  border-bottom-color: #393939; }

.hig-tooltip-placement-right .hig-tooltip-arrow, .hig-tooltip-placement-rightTop .hig-tooltip-arrow, .hig-tooltip-placement-rightBottom .hig-tooltip-arrow {
  left: 4px;
  margin-top: -5px;
  border-width: 5px 5px 5px 0;
  border-right-color: #393939; }

.hig-tooltip-placement-left .hig-tooltip-arrow, .hig-tooltip-placement-leftTop .hig-tooltip-arrow, .hig-tooltip-placement-leftBottom .hig-tooltip-arrow {
  right: 4px;
  margin-top: -5px;
  border-width: 5px 0 5px 5px;
  border-left-color: #393939; }

.hig-tooltip-placement-top .hig-tooltip-arrow {
  left: 50%; }

.hig-tooltip-placement-topLeft .hig-tooltip-arrow {
  left: 15%; }

.hig-tooltip-placement-topRight .hig-tooltip-arrow {
  right: 15%; }

.hig-tooltip-placement-right .hig-tooltip-arrow {
  top: 50%; }

.hig-tooltip-placement-rightTop .hig-tooltip-arrow {
  top: 15%;
  margin-top: 0; }

.hig-tooltip-placement-rightBottom .hig-tooltip-arrow {
  bottom: 15%; }

.hig-tooltip-placement-left .hig-tooltip-arrow {
  top: 50%; }

.hig-tooltip-placement-leftTop .hig-tooltip-arrow {
  top: 15%;
  margin-top: 0; }

.hig-tooltip-placement-leftBottom .hig-tooltip-arrow {
  bottom: 15%; }

.hig-tooltip-placement-bottom .hig-tooltip-arrow {
  left: 50%; }

.hig-tooltip-placement-bottomLeft .hig-tooltip-arrow {
  left: 15%; }

.hig-tooltip-placement-bottomRight .hig-tooltip-arrow {
  right: 15%; }

.hig-tooltip-light.hig-tooltip-placement-top .hig-tooltip-arrow, .hig-tooltip-light.hig-tooltip-placement-topLeft .hig-tooltip-arrow, .hig-tooltip-light.hig-tooltip-placement-topRight .hig-tooltip-arrow {
  border-bottom: none;
  border-top-color: #d4dbe1; }

.hig-tooltip-light.hig-tooltip-placement-right .hig-tooltip-arrow, .hig-tooltip-light.hig-tooltip-placement-rightTop .hig-tooltip-arrow, .hig-tooltip-light.hig-tooltip-placement-rightBottom .hig-tooltip-arrow {
  border-left: none;
  border-right-color: #d4dbe1; }

.hig-tooltip-light.hig-tooltip-placement-left .hig-tooltip-arrow, .hig-tooltip-light.hig-tooltip-placement-leftTop .hig-tooltip-arrow, .hig-tooltip-light.hig-tooltip-placement-leftBottom .hig-tooltip-arrow {
  border-right: none;
  border-left-color: #d4dbe1; }

.hig-tooltip-light.hig-tooltip-placement-bottom .hig-tooltip-arrow, .hig-tooltip-light.hig-tooltip-placement-bottomLeft .hig-tooltip-arrow, .hig-tooltip-light.hig-tooltip-placement-bottomRight .hig-tooltip-arrow {
  border-top: none;
  border-bottom-color: #d4dbe1; }
