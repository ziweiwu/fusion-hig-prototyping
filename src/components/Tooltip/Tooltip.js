import React from "react";
import PropTypes from "prop-types";
import ReactToolTip from "rc-tooltip";
import { polyfill } from "react-lifecycles-compat";
import { anchorPoints } from "@hig/flyout";
import { css } from "react-emotion";
import "./tooltip.scss";

const placements = {
  [anchorPoints.TOP_CENTER]: "top",
  [anchorPoints.TOP_LEFT]: "topLeft",
  [anchorPoints.TOP_RIGHT]: "topRight",
  [anchorPoints.BOTTOM_CENTER]: "bottom",
  [anchorPoints.BOTTOM_LEFT]: "bottomLeft",
  [anchorPoints.BOTTOM_RIGHT]: "bottomRight",
  [anchorPoints.LEFT_CENTER]: "left",
  [anchorPoints.LEFT_TOP]: "leftTop",
  [anchorPoints.LEFT_BOTTOM]: "leftBottom",
  [anchorPoints.RIGHT_CENTER]: "right",
  [anchorPoints.RIGHT_TOP]: "rightTop",
  [anchorPoints.RIGHT_BOTTOM]: "rightBottom"
};

class Tooltip extends React.Component {
  static propTypes = {
    anchorPoint: PropTypes.string,
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
    content: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    description: PropTypes.string,
    id: PropTypes.string,
    isVisible: PropTypes.bool,
    linkTitle: PropTypes.string,
    linkURL: PropTypes.string,
    mouseEnterDelay: PropTypes.number,
    mouseLeaveDelay: PropTypes.number,
    title: PropTypes.string,
    trigger: PropTypes.string,
    width: PropTypes.number
  };

  static defaultProps = {
    anchorPoint: placements[anchorPoints.BOTTOM_CENTER],
    mouseEnterDelay: 1,
    mouseLeaveDelay: 0.2,
    trigger: "click",
    width: 300
  };

  constructor(props) {
    super(props);
    this.state = {
      isVisible: false
    };
    this.handleClick = this.handleClick.bind(this);
    this.visibilityOn = this.visibilityOn.bind(this);
    this.visibilityOff = this.visibilityOff.bind(this);
    this.renderTitle = this.renderTitle.bind(this);
    this.renderDescription = this.renderDescription.bind(this);
    this.renderContent = this.renderContent.bind(this);
    this.renderURL = this.renderURL.bind(this);
    this.renderContainer = this.renderContainer.bind(this);
    this.renderChildren = this.renderChildren.bind(this);
  }

  componentDidMount() {
    document.addEventListener("mousedown", this.handleClick, false);
  }

  componentWillUnmount() {
    document.removeEventListener("mousedown", this.handleClick, false);
  }

  handleClick(e) {
    if (this.childrenRef && this.childrenRef.contains(e.target)) {
      this.visibilityOn();
    } else if (
      this.containerRef &&
      !this.childrenRef.contains(e.target) &&
      !this.containerRef.contains(e.target)
    ) {
      this.visibilityOff();
    }
  }

  visibilityOn() {
    this.setState({
      isVisible: true
    });
  }

  visibilityOff() {
    this.setState({
      isVisible: false
    });
  }

  renderTitle() {
    return <div className="hig__tooltip-title">{this.props.title}</div>;
  }

  renderDescription() {
    return (
      <div className="hig__tooltip-description">{this.props.description}</div>
    );
  }

  renderContent() {
    return (
      <div className="hig__tooltip-inner-content">{this.props.content}</div>
    );
  }

  renderURL() {
    return (
      <div className="hig__tooltip-link">
        {!this.props.content && <hr />}
        <a href={this.props.linkURL}>
          {this.props.linkTitle || this.props.linkURL}
        </a>
      </div>
    );
  }

  renderContainer() {
    return (
      <div
        className="hig__tooltip-container"
        ref={containerRef => (this.containerRef = containerRef)}
      >
        {this.props.title && this.renderTitle()}
        {this.props.description && this.renderDescription()}
        {this.props.content && this.renderContent()}
        {this.props.linkURL && this.renderURL()}
      </div>
    );
  }

  renderChildren() {
    return (
      <div
        className="hig__tooltip-children-wrapper"
        ref={childrenRef => (this.childrenRef = childrenRef)}
      >
        {this.props.children}
      </div>
    );
  }

  render() {
    const { anchorPoint, ...otherProps } = this.props;

    const tooltipWidth = css`
      max-width: ${this.props.width}px;
    `;

    const isVisible =
      this.props.isVisible !== undefined
        ? this.props.isVisible
        : this.state.isVisible;

    return (
      <ReactToolTip
        {...otherProps}
        destroyTooltipOnHide
        overlay={this.renderContainer}
        overlayClassName={tooltipWidth}
        placement={placements[anchorPoint]}
        prefixCls="hig__tooltip"
        transitionName="fade"
        visible={isVisible}
      >
        {this.renderChildren()}
      </ReactToolTip>
    );
  }
}

export default polyfill(Tooltip);
