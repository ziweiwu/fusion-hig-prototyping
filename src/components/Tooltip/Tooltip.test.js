import React from "react";
import { shallow } from "enzyme";
import Tooltip from "./index";

describe("DatePicker", () => {
  // snapshot testing
  test("Snapshot tests right replacement", () => {
    const wrapper = shallow(
      <Tooltip title="test" anchorPoint="right" trigger="click">
        <button>tooltip</button>
      </Tooltip>
    );
    wrapper.simulate("click");
    expect(wrapper).toMatchSnapshot();
  });

  test("Snapshot tests left replacement", () => {
    const wrapper = shallow(
      <Tooltip title="test" anchorPoint="left" trigger="click">
        <button>tooltip</button>
      </Tooltip>
    );
    wrapper.simulate("click");
    expect(wrapper).toMatchSnapshot();
  });

  test("Snapshot tests hover", () => {
    const wrapper = shallow(
      <Tooltip
        title="test"
        anchorPoint="left"
        trigger="hover"
        mouseEnterDelay={0}
      >
        <button>tooltip</button>
      </Tooltip>
    );
    wrapper.simulate("hover");
    expect(wrapper).toMatchSnapshot();
  });

  test("Snapshot tests focus", () => {
    const wrapper = shallow(
      <Tooltip title="test" anchorPoint="left" trigger="focus">
        <button>tooltip</button>
      </Tooltip>
    );
    wrapper.simulate("hover");
    expect(wrapper).toMatchSnapshot();
  });

  // test interactions
});
