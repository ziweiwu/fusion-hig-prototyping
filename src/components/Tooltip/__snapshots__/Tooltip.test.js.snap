// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Snapshot tests focus 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Tooltip
    align={Object {}}
    anchorPoint="left"
    arrowContent={null}
    destroyTooltipOnHide={true}
    mouseEnterDelay={1}
    mouseLeaveDelay={0.2}
    overlay={
        <div
            className="css-d70g0l"
        >
            <div
                className="hig-tooltip-title"
            >
                test
            </div>
        </div>
    }
    placement="right"
    prefixCls="hig-tooltip"
    title="test"
    trigger="focus"
    width={300}
>
    <div
        className="hig-tooltip-children-wrapper"
    >
        <button>
            tooltip
        </button>
    </div>
</Tooltip>,
  "nodes": Array [
    <Tooltip
      align={Object {}}
      anchorPoint="left"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.2}
      overlay={
            <div
                  className="css-d70g0l"
            >
                  <div
                        className="hig-tooltip-title"
                  >
                        test
                  </div>
            </div>
      }
      placement="right"
      prefixCls="hig-tooltip"
      title="test"
      trigger="focus"
      width={300}
>
      <div
            className="hig-tooltip-children-wrapper"
      >
            <button>
                  tooltip
            </button>
      </div>
</Tooltip>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Tooltip
      anchorPoint="left"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.2}
      prefixCls="hig-tooltip"
      title="test"
      trigger="focus"
      width={300}
>
      <button>
            tooltip
      </button>
</Tooltip>,
    "_forcedUpdate": false,
    "_instance": Tooltip {
      "context": Object {},
      "props": Object {
        "anchorPoint": "left",
        "arrowContent": null,
        "children": <button>
          tooltip
</button>,
        "destroyTooltipOnHide": true,
        "mouseEnterDelay": 1,
        "mouseLeaveDelay": 0.2,
        "prefixCls": "hig-tooltip",
        "title": "test",
        "trigger": "focus",
        "width": 300,
      },
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <Tooltip
      align={Object {}}
      anchorPoint="left"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.2}
      overlay={
            <div
                  className="css-d70g0l"
            >
                  <div
                        className="hig-tooltip-title"
                  >
                        test
                  </div>
            </div>
      }
      placement="right"
      prefixCls="hig-tooltip"
      title="test"
      trigger="focus"
      width={300}
>
      <div
            className="hig-tooltip-children-wrapper"
      >
            <button>
                  tooltip
            </button>
      </div>
</Tooltip>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Tooltip
    anchorPoint="left"
    arrowContent={null}
    destroyTooltipOnHide={true}
    mouseEnterDelay={1}
    mouseLeaveDelay={0.2}
    prefixCls="hig-tooltip"
    title="test"
    trigger="focus"
    width={300}
>
    <button>
        tooltip
    </button>
</Tooltip>,
}
`;

exports[`DatePicker Snapshot tests hover 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Tooltip
    align={Object {}}
    anchorPoint="left"
    arrowContent={null}
    destroyTooltipOnHide={true}
    mouseEnterDelay={0}
    mouseLeaveDelay={0.2}
    overlay={
        <div
            className="css-d70g0l"
        >
            <div
                className="hig-tooltip-title"
            >
                test
            </div>
        </div>
    }
    placement="right"
    prefixCls="hig-tooltip"
    title="test"
    trigger="hover"
    width={300}
>
    <div
        className="hig-tooltip-children-wrapper"
    >
        <button>
            tooltip
        </button>
    </div>
</Tooltip>,
  "nodes": Array [
    <Tooltip
      align={Object {}}
      anchorPoint="left"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={0}
      mouseLeaveDelay={0.2}
      overlay={
            <div
                  className="css-d70g0l"
            >
                  <div
                        className="hig-tooltip-title"
                  >
                        test
                  </div>
            </div>
      }
      placement="right"
      prefixCls="hig-tooltip"
      title="test"
      trigger="hover"
      width={300}
>
      <div
            className="hig-tooltip-children-wrapper"
      >
            <button>
                  tooltip
            </button>
      </div>
</Tooltip>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Tooltip
      anchorPoint="left"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={0}
      mouseLeaveDelay={0.2}
      prefixCls="hig-tooltip"
      title="test"
      trigger="hover"
      width={300}
>
      <button>
            tooltip
      </button>
</Tooltip>,
    "_forcedUpdate": false,
    "_instance": Tooltip {
      "context": Object {},
      "props": Object {
        "anchorPoint": "left",
        "arrowContent": null,
        "children": <button>
          tooltip
</button>,
        "destroyTooltipOnHide": true,
        "mouseEnterDelay": 0,
        "mouseLeaveDelay": 0.2,
        "prefixCls": "hig-tooltip",
        "title": "test",
        "trigger": "hover",
        "width": 300,
      },
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <Tooltip
      align={Object {}}
      anchorPoint="left"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={0}
      mouseLeaveDelay={0.2}
      overlay={
            <div
                  className="css-d70g0l"
            >
                  <div
                        className="hig-tooltip-title"
                  >
                        test
                  </div>
            </div>
      }
      placement="right"
      prefixCls="hig-tooltip"
      title="test"
      trigger="hover"
      width={300}
>
      <div
            className="hig-tooltip-children-wrapper"
      >
            <button>
                  tooltip
            </button>
      </div>
</Tooltip>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Tooltip
    anchorPoint="left"
    arrowContent={null}
    destroyTooltipOnHide={true}
    mouseEnterDelay={0}
    mouseLeaveDelay={0.2}
    prefixCls="hig-tooltip"
    title="test"
    trigger="hover"
    width={300}
>
    <button>
        tooltip
    </button>
</Tooltip>,
}
`;

exports[`DatePicker Snapshot tests left replacement 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Tooltip
    align={Object {}}
    anchorPoint="left"
    arrowContent={null}
    destroyTooltipOnHide={true}
    mouseEnterDelay={1}
    mouseLeaveDelay={0.2}
    overlay={
        <div
            className="css-d70g0l"
        >
            <div
                className="hig-tooltip-title"
            >
                test
            </div>
        </div>
    }
    placement="right"
    prefixCls="hig-tooltip"
    title="test"
    trigger="click"
    width={300}
>
    <div
        className="hig-tooltip-children-wrapper"
    >
        <button>
            tooltip
        </button>
    </div>
</Tooltip>,
  "nodes": Array [
    <Tooltip
      align={Object {}}
      anchorPoint="left"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.2}
      overlay={
            <div
                  className="css-d70g0l"
            >
                  <div
                        className="hig-tooltip-title"
                  >
                        test
                  </div>
            </div>
      }
      placement="right"
      prefixCls="hig-tooltip"
      title="test"
      trigger="click"
      width={300}
>
      <div
            className="hig-tooltip-children-wrapper"
      >
            <button>
                  tooltip
            </button>
      </div>
</Tooltip>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Tooltip
      anchorPoint="left"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.2}
      prefixCls="hig-tooltip"
      title="test"
      trigger="click"
      width={300}
>
      <button>
            tooltip
      </button>
</Tooltip>,
    "_forcedUpdate": false,
    "_instance": Tooltip {
      "context": Object {},
      "props": Object {
        "anchorPoint": "left",
        "arrowContent": null,
        "children": <button>
          tooltip
</button>,
        "destroyTooltipOnHide": true,
        "mouseEnterDelay": 1,
        "mouseLeaveDelay": 0.2,
        "prefixCls": "hig-tooltip",
        "title": "test",
        "trigger": "click",
        "width": 300,
      },
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <Tooltip
      align={Object {}}
      anchorPoint="left"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.2}
      overlay={
            <div
                  className="css-d70g0l"
            >
                  <div
                        className="hig-tooltip-title"
                  >
                        test
                  </div>
            </div>
      }
      placement="right"
      prefixCls="hig-tooltip"
      title="test"
      trigger="click"
      width={300}
>
      <div
            className="hig-tooltip-children-wrapper"
      >
            <button>
                  tooltip
            </button>
      </div>
</Tooltip>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Tooltip
    anchorPoint="left"
    arrowContent={null}
    destroyTooltipOnHide={true}
    mouseEnterDelay={1}
    mouseLeaveDelay={0.2}
    prefixCls="hig-tooltip"
    title="test"
    trigger="click"
    width={300}
>
    <button>
        tooltip
    </button>
</Tooltip>,
}
`;

exports[`DatePicker Snapshot tests right replacement 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Tooltip
    align={Object {}}
    anchorPoint="right"
    arrowContent={null}
    destroyTooltipOnHide={true}
    mouseEnterDelay={1}
    mouseLeaveDelay={0.2}
    overlay={
        <div
            className="css-d70g0l"
        >
            <div
                className="hig-tooltip-title"
            >
                test
            </div>
        </div>
    }
    placement="right"
    prefixCls="hig-tooltip"
    title="test"
    trigger="click"
    width={300}
>
    <div
        className="hig-tooltip-children-wrapper"
    >
        <button>
            tooltip
        </button>
    </div>
</Tooltip>,
  "nodes": Array [
    <Tooltip
      align={Object {}}
      anchorPoint="right"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.2}
      overlay={
            <div
                  className="css-d70g0l"
            >
                  <div
                        className="hig-tooltip-title"
                  >
                        test
                  </div>
            </div>
      }
      placement="right"
      prefixCls="hig-tooltip"
      title="test"
      trigger="click"
      width={300}
>
      <div
            className="hig-tooltip-children-wrapper"
      >
            <button>
                  tooltip
            </button>
      </div>
</Tooltip>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Tooltip
      anchorPoint="right"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.2}
      prefixCls="hig-tooltip"
      title="test"
      trigger="click"
      width={300}
>
      <button>
            tooltip
      </button>
</Tooltip>,
    "_forcedUpdate": false,
    "_instance": Tooltip {
      "context": Object {},
      "props": Object {
        "anchorPoint": "right",
        "arrowContent": null,
        "children": <button>
          tooltip
</button>,
        "destroyTooltipOnHide": true,
        "mouseEnterDelay": 1,
        "mouseLeaveDelay": 0.2,
        "prefixCls": "hig-tooltip",
        "title": "test",
        "trigger": "click",
        "width": 300,
      },
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <Tooltip
      align={Object {}}
      anchorPoint="right"
      arrowContent={null}
      destroyTooltipOnHide={true}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.2}
      overlay={
            <div
                  className="css-d70g0l"
            >
                  <div
                        className="hig-tooltip-title"
                  >
                        test
                  </div>
            </div>
      }
      placement="right"
      prefixCls="hig-tooltip"
      title="test"
      trigger="click"
      width={300}
>
      <div
            className="hig-tooltip-children-wrapper"
      >
            <button>
                  tooltip
            </button>
      </div>
</Tooltip>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Tooltip
    anchorPoint="right"
    arrowContent={null}
    destroyTooltipOnHide={true}
    mouseEnterDelay={1}
    mouseLeaveDelay={0.2}
    prefixCls="hig-tooltip"
    title="test"
    trigger="click"
    width={300}
>
    <button>
        tooltip
    </button>
</Tooltip>,
}
`;
